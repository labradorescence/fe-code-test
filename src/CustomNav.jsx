import React from "react";

import logo from "./logo.jpg";
import title from "./TNRTitle.png";

const CustomNav = __ => (
  <nav className="nav">
    <a className="skipNav" href="#main">
      skip navigation{" "}
    </a>
    <i className="fas fa-bars fa-2x hamburgerIcon" />
    <img className="logo" alt="TNR logo" src={logo} />
    <img className="header" alt="The New Republic" src={title} />
    <a
      href="https://newrepublic.com/magazine"
      className="nav__link nav__link--mag"
    >
      {" "}
      MAGAZINE{" "}
    </a>
    <a href="/imaginary-subscription-page" className="nav__link nav__link--sub">
      {" "}
      SUBSCRIBE{" "}
    </a>
  </nav>
);

export default CustomNav