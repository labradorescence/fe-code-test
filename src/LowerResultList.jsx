import React from "react";
import PropTypes from "prop-types";
import DOMPurify from "dompurify";

const LowerResultList = props => {
  return (
    <span>
      {props.articles.length ? (
        <span className="lowerResultList__title">MOST POPULAR</span>
      ) : null}
      <ul className="lowerResultList">
        {props.articles.map((article, ind) => (
          <li className="searchCard searchCard--rest" key={ind}>
            <a className="searchCard--restLink" href={article.urlFull}>
              {article.ledeImage ? (
                <img
                  className="lowerResult__image"
                  src={article.ledeImage.src}
                  alt={
                    article.ledeAltImage || "this needs a descriptive alt tag!"
                  }
                />
              ) : (
                <img
                  alt="filler kitten"
                  className="lowerResult__image"
                  src="http://placekitten.com/g/200/200"
                />
              )}
              <span className="lowerResult__text">
                <span
                  className="lowerResult__title"
                  dangerouslySetInnerHTML={{
                    __html: DOMPurify.sanitize(article.title)
                  }}
                />
                <span className="lowerResult__author">
                  {article.authors.map((author, ind) =>
                    ind < article.authors.length - 1
                      ? author.name + ", "
                      : author.name
                  )}
                </span>
              </span>
            </a>
          </li>
        ))}
      </ul>
    </span>
  );
};

LowerResultList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      nid: PropTypes.number,
      title: PropTypes.string,
      slug: PropTypes.string,
      publishedAt: PropTypes.string,
      url: PropTypes.string,
      authors: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string,
          slug: PropTypes.string
        })
      ),
      ledeImage: PropTypes.shape({
        src: PropTypes.string,
        width: PropTypes.number,
        height: PropTypes.number
      })
    })
  )
};

export default LowerResultList;