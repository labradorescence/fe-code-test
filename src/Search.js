import React from "react"
import SearchResults from "./SearchResults.jsx"


class Search extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      value: ''
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({value: event.target.value});
  }
  



  render() {

    return <div>

      <div className="above-result-container" id="above-result">
        <section className="above-result">
        
        <span className="search">
        Search: <input className="search-input" type="text" value={this.state.value} onChange={this.handleChange}/></span>
        </section>
      </div>



      <SearchResults term={this.state.value} />

    </div>
  }
}

export default Search
