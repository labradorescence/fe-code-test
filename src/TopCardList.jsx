import React from "react";
import PropTypes from "prop-types";
import DOMPurify from "dompurify";

const TopCardList = props => (
  <ul className="topResults">
    {props.articles.map((article, ind) => (
      <li className="searchCard searchCard--topThree" key={ind}>
        <a href={article.urlFull}>
          {article.ledeImage ? (
            <img
              className="searchCard__image"
              src={article.ledeImage.src}
              alt={article.ledeAltImage || "this needs a descriptive alt tag!"}
            />
          ) : (
            <img
              alt="filler kitten"
              className="searchCard__image"
              src="http://placekitten.com/g/200/200"
            />
          )}
          <span className="topResults__year">
            {article.publishedAt.slice(0, 4)}
          </span>
          <h4
            dangerouslySetInnerHTML={{
              __html: DOMPurify.sanitize(article.title)
            }}
            className="topResults__title"
          />
          <span className="topResults__author">
            {article.authors.map(author => author.name)}
          </span>
        </a>
      </li>
    ))}
  </ul>
);

TopCardList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      nid: PropTypes.number,
      title: PropTypes.string,
      slug: PropTypes.string,
      publishedAt: PropTypes.string,
      url: PropTypes.string,
      authors: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string,
          slug: PropTypes.string
        })
      ),
      ledeImage: PropTypes.shape({
        src: PropTypes.string,
        width: PropTypes.number,
        height: PropTypes.number
      })
    })
  )
};

export default TopCardList;
