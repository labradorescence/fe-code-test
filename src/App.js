import React from 'react';
import Search from "./Search";
import Nav from "./Nav.jsx";

const App = () => 
    <div>
    <Nav/>
    <Search/>
    </div>

export default App;
