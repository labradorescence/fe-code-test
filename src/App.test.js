import React from "react";
import ReactDOM from "react-dom";

import { mount, configure, shallow } from "enzyme";
import { render } from "@testing-library/react";

import CustomNav from "./CustomNav";
import TopCardList from "./TopCardList";
import LowerResultList from "./LowerResultList";
import Search from './Search'
import App from "./App";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


/*
	Normally, I'd want to add more thorough tests, 
	but given the time constraints, I focused on 
	getting the essential minimum of coverage with a suite of 
	snapshot tests to keep track of changes
*/

describe("snapshot suite", () => {
  it("CustomNav looks right", () => {
    const { container } = render(<CustomNav />);
    expect(container).toMatchSnapshot();
  });
   it("TopCardList looks right", () => {
    const { container } = render(<TopCardList articles={[]}/>);
    expect(container).toMatchSnapshot();
  });
  it("LowerResultList looks right", () => {
    const { container } = render(<LowerResultList  articles={[]} />);
    expect(container).toMatchSnapshot();
  });
  it("Search looks right", () => {
    const { container } = render(<Search />);
    expect(container).toMatchSnapshot();
  });
});