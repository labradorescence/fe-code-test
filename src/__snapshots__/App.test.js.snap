// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot suite CustomNav looks right 1`] = `
<div>
  <nav
    class="nav"
  >
    <a
      class="skipNav"
      href="#main"
    >
      skip navigation
       
    </a>
    <i
      class="fas fa-bars fa-2x hamburgerIcon"
    />
    <img
      alt="TNR logo"
      class="logo"
      src="logo.jpg"
    />
    <img
      alt="The New Republic"
      class="header"
      src="TNRTitle.png"
    />
    <a
      class="nav__link nav__link--mag"
      href="https://newrepublic.com/magazine"
    >
       
      MAGAZINE
       
    </a>
    <a
      class="nav__link nav__link--sub"
      href="/imaginary-subscription-page"
    >
       
      SUBSCRIBE
       
    </a>
  </nav>
</div>
`;

exports[`snapshot suite LowerResultList looks right 1`] = `
<div>
  <span>
    <ul
      class="lowerResultList"
    />
  </span>
</div>
`;

exports[`snapshot suite Search looks right 1`] = `
<div>
  <div>
    <nav
      class="nav"
    >
      <a
        class="skipNav"
        href="#main"
      >
        skip navigation
         
      </a>
      <i
        class="fas fa-bars fa-2x hamburgerIcon"
      />
      <img
        alt="TNR logo"
        class="logo"
        src="logo.jpg"
      />
      <img
        alt="The New Republic"
        class="header"
        src="TNRTitle.png"
      />
      <a
        class="nav__link nav__link--mag"
        href="https://newrepublic.com/magazine"
      >
         
        MAGAZINE
         
      </a>
      <a
        class="nav__link nav__link--sub"
        href="/imaginary-subscription-page"
      >
         
        SUBSCRIBE
         
      </a>
    </nav>
    <main
      id="main"
    >
      <span
        class="search"
      >
        <label
          class="search__label"
          for="search-input"
        >
           
          Search:
           
        </label>
        <input
          class="search__input"
          id="search-input"
          type="text"
          value=""
        />
      </span>
      <div>
        <span />
        <ul
          class="topResults"
        />
        <span>
          <ul
            class="lowerResultList"
          />
        </span>
      </div>
    </main>
  </div>
</div>
`;

exports[`snapshot suite TopCardList looks right 1`] = `
<div>
  <ul
    class="topResults"
  />
</div>
`;
